.github/workflows/linux.yml

```
@@ -0,0 +1,84 @@
+# Copyright (c) 2023 Sebastian Pipping <sebastian@pipping.org>
+# Licensed under GPL v2 or later
+
+name: Build on Linux
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: '0 3 * * 5'  # Every Friday at 3am
+  workflow_dispatch:
+
+jobs:
+  linux:
+    name: Build (${{ matrix.cc }})
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - cc: gcc-13
+            cxx: g++-13
+            clang_major_version: null
+            clang_repo_suffix: null
+            runs-on: ubuntu-22.04
+          - cc: clang-17
+            cxx: clang++-17
+            clang_major_version: 17
+            clang_repo_suffix: -17
+            runs-on: ubuntu-22.04
+          - cc: clang-18
+            cxx: clang++-18
+            clang_major_version: 18
+            clang_repo_suffix:
+            runs-on: ubuntu-22.04
+    steps:
+      - name: Add Clang/LLVM repositories
+        if: "${{ contains(matrix.cxx, 'clang') }}"
+        run: |-
+          set -x
+          source /etc/os-release
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo add-apt-repository "deb http://apt.llvm.org/${UBUNTU_CODENAME}/ llvm-toolchain-${UBUNTU_CODENAME}${{ matrix.clang_repo_suffix }} main"
+
+      - name: Install build dependencies
+        run: |-
+          sudo apt-get update
+          sudo apt-get install --yes --no-install-recommends \
+            autoconf \
+            automake \
+            libfuse3-dev \
+            libtool \
+            pkg-config
+
+      - name: Install build dependency Clang ${{ matrix.clang_major_version }}
+        if: "${{ contains(matrix.cxx, 'clang') }}"
+        run: |-
+          sudo apt-get install --yes --no-install-recommends -V \
+              clang-${{ matrix.clang_major_version }}
+
+      - name: Checkout Git branch
+        uses: actions/checkout@v4
+
+      - name: 'Bootstrap with ./autogen.sh'
+        run: |-
+          ./autogen.sh
+
+      - name: 'Configure'
+        run: |-
+          set -x
+          mkdir build
+          cd build
+          ../configure --with-fuse3
```

```suggestion
          ../configure
```

It uses fuse3 when available automatically. By removing the flag, the command will also work on older distros.

---

@mpartel auto-detection, understood. Due to https://github.com/mpartel/bindfs/pull/146#discussion_r1390237554 I'm not sure how much other Linuxes we will have at that layer and package `libfuse3-dev` is installed explicitly. I felt that being explicit would help, if you still want me to drop it, please let me know.
PS: One could cover all sorts of combinations for `--with-fuse3` and `--with-fuse2` and without and both packages installed or none and so one, but things get expensive quickly and the gain will likely have a hard time keeping up.

---

@mpartel PS: dropped `--with-fuse3` now as suggested.

---

Might be worth adding fuse2/fuse3 to the matrix, if possible. But that's for another commit.